elide-model-config/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -343,16 +337,77 @@ private static void validateConfigForMissingVariables(String config, Map<String,
     private static boolean validateTableConfig(ElideTableConfig elideTableConfig) {
         for (Table table : elideTableConfig.getTables()) {
             validateSql(table.getSql());
-            table.getDimensions().forEach(dim -> validateSql(dim.getDefinition()));
+            table.getDimensions().forEach(dim -> {
+                validateSql(dim.getDefinition());
+                validateTableSource(elideTableConfig, dim.getTableSource());
+            });
             table.getMeasures().forEach(measure -> validateSql(measure.getDefinition()));
             table.getJoins().forEach(join -> validateJoin(join, elideTableConfig));
         }
+
         return true;
     }
 
+    private static void validateTableSource(ElideTableConfig elideTableConfig, String tableSource) {
+        if (isNullOrEmpty(tableSource)) {
+            return; // Nothing to validate
+        }
+
+        String[] split = tableSource.split("\\.");
+        if (split.length != 2) {
+            throw new IllegalStateException("Invalid tableSource: " + tableSource
+                            + ". tableSource must be in format: modelName.columnName");
+        }
+        String modelName = split[0];
+        String fieldName = split[1];
+
+        if (!staticModelContainsField(modelName, NO_VERSION, fieldName)
+                        && !dynamicModelContainsField(elideTableConfig, modelName, fieldName)) {
+            throw new IllegalStateException("Invalid tableSource : " + tableSource + " . Either model : " + modelName
+                            + " is undefined or field : " + fieldName + " is undefined for this model.");
```

Comment:
We should break these errors out to help the developer.

Reply:
sure.